src/cargo/util/toml.rs

```diff
@@ -1040,7 +1051,11 @@ fn normalize(lib: &Option<TomlLibTarget>,
               .set_doctest(toml.doctest.unwrap_or(t2.doctested()))
               .set_benched(toml.bench.unwrap_or(t2.benched()))
               .set_harness(toml.harness.unwrap_or(t2.harness()))
-              .set_for_host(toml.plugin.unwrap_or(t2.for_host()));
+              .set_for_host(match (toml.plugin, toml.rustc_macro) {
+                  (None, None) => t2.for_host(),
+                  (Some(true), _) | (_, Some(true)) => true,
+                  _ => false,
```

Comment:
Well out of the 9 cases, 1 is handled by the first line and 5 are handled by the second line so there are 3 left:

```rust
(None, Some(false))
(Some(false), None)
(Some(false), Some(false))
```

Are you suggesting this?

```rust
(Some(false), _) | (_, Some(false)) => false
```

I think the current code is clearer. It says "if nothing is set, inherit; if anything is true, then true otherwise false." The explicitly bound `Some(false)` looks like it means "if anything is false, then false" but that is misleading because `(Some(true), Some(false))` is true.